fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1755,6 +1783,19 @@ private static QueryParameter parseChainedParameter(Class<?> resourceType, Strin
 
             List<QueryParameterValue> valueList =
                     processQueryParameterValueString(resourceType, searchParameter, modifier, rootParameter.getModifierResourceTypeName(), valuesString);
+
+            if (checkForLogicalId) {
+                // For last search parameter, if type REFERENCE and not scoped to single target resource type, check if
+                // value is logical ID only.
+                for (QueryParameterValue value : valueList) {
+                    ReferenceValue refVal = ReferenceUtil.createReferenceValueFrom(value.getValueString(), null, ReferenceUtil.getBaseUrl(null));
+                    if (refVal.getType() == ReferenceType.LITERAL_RELATIVE && refVal.getTargetResourceType() == null) {
+                        throw SearchExceptionUtil.buildNewInvalidSearchException(
+                            String.format(LOGICAL_ID_VALUE_NOT_ALLOWED_FOR_REFERENCE_SEARCH, rootParameter.getChain().getLast().getCode(), value.getValueString()));
+                    }
+                }
```

Comment:
You may want to refactor this into a new method. It's very similar to the prior method.

Reply:
Refactored into new method.